homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -209,3 +210,45 @@ async def async_unload_entry(
         return False
 
     return True
+
+
+async def async_migrate_entry(
+    hass: HomeAssistant, entry: GoogleGenerativeAIConfigEntry
+) -> bool:
+    """Migrate old entry."""
+    if entry.version == 1:
+        # Migrate from version 1 to version 2
+        # Move conversation-specific options to a subentry
+        subentry = ConfigSubentry(
+            data=entry.options,
+            subentry_type="conversation",
+            title=DEFAULT_CONVERSATION_NAME,
+            unique_id=None,
+        )
+        hass.config_entries.async_add_subentry(
+            entry,
+            subentry,
+        )
+
+        # Migrate conversation entity to be linked to subentry
+        ent_reg = er.async_get(hass)
+        conversation_entity = ent_reg.async_get_entity_id(
+            "conversation",
+            DOMAIN,
+            entry.entry_id,
+        )
+        if conversation_entity is not None:
+            ent_reg.async_update_entity(
+                conversation_entity,
+                config_subentry_id=subentry.subentry_id,
```

Comment:
Note to self, can we have an entry_id pointing to an entry and sub entry to a subentry that is not a sub entry of that entry